File: csharp/rna-transcription/RnaTranscription.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class RnaTranscription
{
    private static readonly Dictionary<char, char> transcriptionMap =
        new Dictionary<char, char>
        {
            ['G'] = 'C',
            ['C'] = 'G',
            ['T'] = 'A',
            ['A'] = 'U'
        };

    public static string ToRna(string nucleotide) =>
        String.Concat(nucleotide.Select(n => transcriptionMap[n]));
}

File: fsharp/rna-transcription/RnaTranscription.fs
module RnaTranscription

open System

let toRna (dna: string): string =
    let transcribe = function
    | 'G' -> 'C'
    | 'C' -> 'G'
    | 'T' -> 'A'
    | 'A' -> 'U'
    | _ -> Char.MinValue

    dna |> Seq.map transcribe |> String.Concat